tools/GenerateDocFxStructure/Program.cs

```diff
@@ -0,0 +1,313 @@
+        private static void ProcessFiles(DirectoryInfo folder)
+        {
+            _message!.Verbose($"Process {folder.FullName} for files.");
+
+            List<FileInfo> files = folder.GetFiles("*.md").OrderBy(f => f.Name).ToList();
+            if (files == null)
+            {
+                _message.Verbose($"No MD files found in {folder.FullName}.");
+                return;
+            }
+
+            foreach (FileInfo fi in files)
+            {
+                _message.Verbose($"Processing {fi.FullName}");
+                string content = File.ReadAllText(fi.FullName);
+
+                // first see if there are links in this file
+                Regex rxContent = new Regex(@"(\[{1}.*\]{1}\({1}.*\){1}?)");
+                if (rxContent.Matches(content).Any())
+                {
+                    _message.Verbose($"- Links detected.");
+
+                    // it has references, so check in detail
+                    ProcessFile(folder, fi.FullName);
+                }
+                else
+                {
+                    // Create the destination file
+                    var relativeDestinationFilePath = fi.FullName.Substring(_options!.SourceFolder!.Length + 1);
+                    var newFileDestination = Path.Combine(_options.DestinationFolder!, relativeDestinationFilePath);
+                    // Ensure the path exists and is created
+                    var dir = Path.GetDirectoryName(newFileDestination);
+                    if (!Directory.Exists(dir))
+                    {
+                        Directory.CreateDirectory(dir!);
+                    }
+
+                    File.Copy(fi.FullName, newFileDestination);
+                }
+
+                _message.Verbose($"{fi.FullName} processed.");
+            }
+        }
+
+        /// <summary>
+        /// Process given file in give folder. Check all references.
+        /// </summary>
+        /// <param name="folder">Folder where file live.</param>
+        /// <param name="filepath">Complete path of the file to check.</param>
+        private static void ProcessFile(DirectoryInfo folder, string filepath)
+        {
+            if ((_options == null) ||(_message == null))
+            {
+                throw new ArgumentNullException($"Options or message can't be null, something went wrong");
+            }
+
+            string[] lines = File.ReadAllLines(filepath);
+            // Create the destination file
+            var relativeDestinationFilePath = filepath.Substring(_options.SourceFolder!.Length + 1);
+            var newFileDestination = Path.Combine(_options.DestinationFolder!, relativeDestinationFilePath);
+            // Ensure the path exists and is created
+            var dir = Path.GetDirectoryName(newFileDestination);
+            if (!Directory.Exists(dir))
+            {
+                Directory.CreateDirectory(dir!);
+            }
+
+            StreamWriter destinationFile = File.CreateText(newFileDestination);
```

Comment:
StringBuilder + File.WriteAllText might be easier here

Reply:
if you stick with StreamWriter use `using` statement and get rid of flush + close